src/frontend/sharedComponents/DeviceNameWithIcon.tsx

```diff
@@ -100,23 +95,59 @@ export const DeviceNameWithIcon = ({
   );
 };
 
-const DeviceArchive = () => {
+const DeviceIcon = ({
+  deviceType,
+  size,
+}: {
+  deviceType: DeviceType | undefined;
+  size: number;
+}) => {
+  switch (deviceType) {
+    case 'mobile':
+      return <DeviceIconWithBackground icon="phone-android" size={size} />;
+    case 'tablet':
+      return <DeviceIconWithBackground icon="tablet-android" size={size} />;
+    case 'desktop':
+      return <DeviceIconWithBackground icon="computer" size={size} />;
+    case 'selfHostedServer':
+      return (
+        <DeviceIconWithBackground
+          icon="shield-lock"
+          size={size}
+          iconLibrary="octicons"
+        />
+      );
+    case undefined:
+    case 'UNRECOGNIZED':
+    case 'device_type_unspecified':
+    default: {
+      return <DeviceIconWithBackground icon="help-outline" size={size} />;
+    }
+  }
+};
+
+const DeviceIconWithBackground = ({
+  icon,
+  size,
+  iconLibrary = 'material',
+}: {
+  icon: string;
+  size: number;
+  iconLibrary?: 'material' | 'octicons';
+}) => {
+  const IconComponent = iconLibrary === 'octicons' ? Octicons : MaterialIcons;
+
   return (
     <View
       style={{
         alignItems: 'center',
-        position: 'relative',
+        justifyContent: 'center',
         backgroundColor: LIGHT_GREY,
-        borderRadius: 100,
-        padding: 40,
-        width: 35,
-        height: 35,
+        borderRadius: size / 2,
+        width: size,
+        height: size,
       }}>
-      <ShieldIcon
-        width={50}
-        height={50}
-        style={{position: 'absolute', top: 15, left: 15}}
-      />
+      <IconComponent name={icon} size={size * 0.5} color={DARK_GREY} />
```

Comment:
non-blocking and somewhat stylistic, but the complicated props interface for this component is usually a sign to consider composition:

```tsx
// Could argue that having this defined as a component isn't even necessary...
const DeviceIconBackground = ({ children, size }) => {
  return (
    <View style={{alignItems: 'center', backgroundColor: LIGHT_GREY, borderRadius: size / 2, width: size, height: size}}>
      {children}
    </View>
  )
})

// In the consuming component
const DeviceIcon = ({size}) => {
  // example of the self hosted server device type case...
  return (
    <DeviceIconBackground size={size}>
      <Octicons name="shield-lock" size={size * 0.5} color={DARK_GREY} />
    </DeviceIconBackground>
  )
}
```